kafka-connect-s3/src/main/java/io/confluent/connect/s3/storage/S3OutputStream.java

```diff
@@ -101,6 +101,9 @@ public void write(byte[] b, int off, int len) throws IOException {
       write(b, off + firstPart, len - firstPart);
     } else {
       buffer.put(b, off, len);
+      if (!buffer.hasRemaining()) {
```

Comment:
That's a great catch @staslos! 
However, I'd prefer to push a slightly different fix, to keep having a single path (as opposed to two) where upload can happen in this method. In the `if` branch I'd rewrite the condition as `if (buffer.remaining() <= len) {`